apps/web/src/components/transactions/TxDetails/TxData/DecodedData/SingleTxDecoded/index.tsx

```diff
@@ -30,6 +40,54 @@ export const SingleTxDecoded = ({ tx, txData, actionTitle, variant, expanded, on
 
   const safeToL2MigrationDeployment = getSafeToL2MigrationDeployment()
   const safeToL2MigrationAddress = chain && safeToL2MigrationDeployment?.networkAddresses[chain.chainId]
+  const tokenInfoIndex = (txData as TransactionDetails['txData'])?.tokenInfoIndex
+
+  const txDataHex = tx.data ?? txData.hexData
+
+  const isTransfer = Boolean(txDataHex?.startsWith(ERC20_INTERFACE.getFunction('transfer').selector))
+
+  const transferInfos:
+    | {
+        recipient: string
+        transferValue: string
+        tokenInfo: Erc20Token | NativeToken
+      }
+    | undefined = useMemo(() => {
+    if ((!isTransfer && !isNativeTransfer) || !txDataHex || !chain) {
```

Comment:
For native transfers, there isn't necessarily `txDataHex` - does this still return them?

Reply:
I changed the logic a bit to only check the hexData for ERC20 transfers and extracted the useMemo into a `useTransferTokenInfo` hook.